File: 9-special-pythagorean-triplet/main.hs
-- Find the product of the only pythagorean triple whose sum is 1000

-- triples is all pythagorean triples that add up to 1000 (there is only one)
triples = [(a, b, c) | b <- [2..], a <- [1..(b-1)], let c = 1000 - a - b, a^2 + b^2 == c^2]

main = do
  let triple = head triples
  print triple
  print (threeTupleProduct triple) where
    threeTupleProduct (a, b, c) = a * b * c

File: 9-special-pythagorean-triplet/main.rb
# Find the pythagorean triplet which adds up to 1000.

def triple a, b, c
  a**2 + b**2 == c**2
end


def triple_totalling n
  c = 5
  loop do
    b = 2
    while b < c
      a = n - b - c
      if triple a, b, c
        return [a, b, c] if a + b + c == n
      end
      b += 1
    end
    c += 1
  end
end

a, b, c = triple_totalling 1000
puts a * b * c
